### PR TITLE
Issue 50 separate imu cam

### PR DESCRIPTION
**Pull request by qinjunj**
Please review this PR with the main ILLIXR PR [ Separate IMU and Cam #324 ](https://github.com/ILLIXR/ILLIXR/pull/324).

### CODE REVIEW

**Review by charmoniumQ — DISMISSED** [review]

**Comment by jianxiapyh** [comment]
Closed and addressed in PR #363

File: ov_msckf/src/slam2.cpp [hunk 1]
```diff
@@ -205,29 +206,40 @@ class slam2 : public plugin {
...
 			return;
-		} else if (imu_cam_buffer == NULL) {
-			imu_cam_buffer = datum;
+		}
+		if (!cam_buffer) {
+			cam_buffer = cam;
+			return;
+		}
+		/* This is necsesary in async, to make sure that cam is new
+		   In buffered, cam is new if it is present.
```
Comment (jianxiapyh) [inline ov_msckf/src/slam2.cpp]: do we still need this

File: ov_msckf/src/slam2.cpp [hunk 2]
```diff
@@ -298,22 +306,23 @@ class slam2 : public plugin {
 		// Turns out, this is no longer correct. debbugview uses it
 		// const_cast<imu_cam_type*>(imu_cam_buffer)->img0.reset();
```
Comment (jianxiapyh) [inline ov_msckf/src/slam2.cpp]: again do we need it?

File: ov_msckf/src/slam2.cpp (hunk 1 above)
Comment (qinjunj) [reply ov_msckf/src/slam2.cpp]: Yes. We do not change anything in OpenVINS and the buffer is to be consistent with OpenVINS. We may not need it when we integrate the latest OpenVINS.

File: ov_msckf/src/slam2.cpp (hunk 2 above)
Comment (qinjunj) [reply ov_msckf/src/slam2.cpp]: same as the last

File: ov_msckf/src/slam2.cpp (hunk 1 above)
Comment (jianxiapyh) [reply ov_msckf/src/slam2.cpp]: I think it is better to have a flag for these. CAMERA_ASYNC something like that

Comment (qinjunj) [reply ov_msckf/src/slam2.cpp]: I think this comment is stale. We are not using async but buffered reader, which guarantees that if a cam is present, it is new. I'll remove the comment.

Comment (jianxiapyh) [reply ov_msckf/src/slam2.cpp]: ok thats what i thought. just remove it in this case.

**Review by jianxiapyh — DISMISSED** [review]
remove stale comment code and its good to go

**Review by power-rdx — DISMISSED** [review]
LGTM

**Review by jianxiapyh — APPROVED** [review]

**Review by power-rdx — APPROVED** [review]

**Pull request merged**